File: backend/app/schemas/director.py
from datetime import date, datetime
from typing import List, Optional

from pydantic import BaseModel


class DirectorOut(BaseModel):
    id: int
    tmdb_id: int
    name: str
    biography: Optional[str] = None
    profile_url: Optional[str] = None
    birthday: Optional[date] = None
    deathday: Optional[date] = None
    place_of_birth: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class DirectorMovieOut(BaseModel):
    id: int
    tmdb_id: int
    title: str
    description: Optional[str] = None
    release_date: Optional[date] = None
    poster_url: Optional[str] = None
    avg_rating: float

    model_config = {"from_attributes": True}


class DirectorProfileOut(DirectorOut):
    movies: List[DirectorMovieOut] = []

File: android/app/src/main/java/com/app/movieit/data/model/Director.kt
package com.app.movieit.data.model

import com.google.gson.annotations.SerializedName

data class Director(
    val id: Int,
    @SerializedName("tmdb_id") val tmdbId: Int,
    val name: String,
    val biography: String? = null,
    @SerializedName("profile_url") val profileUrl: String? = null,
    val birthday: String? = null,
    val deathday: String? = null,
    @SerializedName("place_of_birth") val placeOfBirth: String? = null,
)

data class DirectorMovie(
    val id: Int,
    @SerializedName("tmdb_id") val tmdbId: Int,
    val title: String,
    @SerializedName("release_date") val releaseDate: String? = null,
    @SerializedName("poster_url") val posterUrl: String? = null,
    @SerializedName("avg_rating") val avgRating: Double = 0.0,
)

data class DirectorProfile(
    val id: Int,
    @SerializedName("tmdb_id") val tmdbId: Int,
    val name: String,
    val biography: String? = null,
    @SerializedName("profile_url") val profileUrl: String? = null,
    val birthday: String? = null,
    val deathday: String? = null,
    @SerializedName("place_of_birth") val placeOfBirth: String? = null,
    val movies: List<DirectorMovie> = emptyList(),
)
